docs/src/proposals/vote_subcommittee.md

```diff
@@ -0,0 +1,137 @@
+receive the block, it is possible to allow a sampled set of validators
+to vote on that block as apposed to all the validators, and achieve
+the same liveness properties. If the network is tolerant of 1/3
+failures, then for a large enough random sample, like 200 or more,
+the probability of containing 2/3+ faulty nodes is 1:10^40. Assuming
+at least 1/3+ honest nodes are present in the subcommittee, then
+faulty forks will not get finality, and network will be able to
+continue.
+
+## Detailed Solution
+
+The following sections provide more details of the design.
+
+### Definitions
+
+* Voting Subcommittee: the set of nodes currently voting on blocks
+
+* Voting epoch: the number of slots that the voting subcommittee
+is voting for
+
+* primary subcommittee: The half of the voting subcommittee that
+is scheduled for its second epoch.
+
+* secondary subcommittee: The half of the voting subcommittee that
+is scheduled for its first epoch.
+
+* rotation block: The last block in the previous epoch that was
+rooted by 2/3+ of both primary and secondary super-majorities.
+
+* rotation seed: The seed used to generate the random sample of nodes.
+It's computed as `slow_hash(rotation block bank_hash)`
+
+
+* `slow_hash`: repeated sha256 for rounds equal to 1ms on modern
+sha-ni hardware.
+
+### Safety Violations
+
+An Optimistic Confirmation violation occurs if a voting subcommittee
+optimistically confirms a slot that wasn't rooted by the following
+voting subcommittee.
+
+A Safety violation occurs if a subcommittee roots a slot that wasn't
+rooted by the following subcommittee.
+
+All full nodes including RPCs should halt as soon as a safety
+violation is detected.
+
+### Subcommittee Rotation
+
+Voting Subcommittee is scheduled for a voting epoch, and is itself
+composed of 2 subcommittees, primary and secondary, scheduled for
+two epochs each in a staggered rotation.
+
+```
+A1 A1 A2 A2 A3
+   B1 B1 B2 B2
+```
+
+The first epoch the subcommittee runs, it is considered **secondary**.
+On the second epoch, it is considered **primary**.
+
+For a new secondary subcommittee to start, it must start with a
+block that is a descendant of a **rotation block** that was confirmed
+by both super-majorities in the previous epoch. The **rotation seed**
+is used to derive the new secondary subcommittee.
+
+### Subcommittee Selection
+
+A stake weighted shuffle from the **rotation seed** is used to pick
+the next secondary subcommittee. A slow hash is necessary to ensure
+that grinding the **rotation block** bankhash is not practical. A
+random sample of 200 nodes picked with a stake weighed shuffle.
+Assuming 1/3 are faulty, probability of 133 or greater faulty nodes
+is 1:10^20. A 1ms hash would require grinding for 10^9 years with
+1 billion cores.
+
+### Threshold Switching
+
+Secondary subcommittee can use the primary's votes as a switching
+threshold proof. The primary can only use its own votes for switching
```

Comment:
Can primary use primary committee votes that were submitted from past epoch when it was secondary

Reply:
Only for the fork it created while it was secondary.